pkg/api/v1alpha1/cluster_types.go

```diff
@@ -58,6 +58,18 @@ type ClusterSpec struct {
 	BundlesRef *BundlesRef `json:"bundlesRef,omitempty"`
 }
 
+// HasAWSIamConfig checks if AWSIamConfig is configured for the cluster.
+func (c *Cluster) HasAWSIamConfig() bool {
+	for _, identityProvider := range c.Spec.IdentityProviderRefs {
+		switch identityProvider.Kind {
+		case AWSIamConfigKind:
+			return true
+		}
```

Comment:
why a switch for only one case?
```suggestion
		if identityProvider.Kind == AWSIamConfigKind {
			return true
		}
```